### PR TITLE
Switch email provider to Mailgun. Redirect to dashboard on login if user has a valid session

### PR DESCRIPTION
**Pull request by andresgutgon**
# What?
☝️ That

### TODO
- [x] Deploy infra environment

## Issue 
https://github.com/latitude-dev/latitude-llm/issues/444

### CODE REVIEW

File: packages/env/src/index.ts [hunk 1]
```diff
@@ -70,6 +71,7 @@ export const env = createEnv({
     FROM_MAILER_EMAIL: z.string(),
     GATEWAY_HOSTNAME: z.string(),
     LATITUDE_DOMAIN: z.string(),
+    LATITUDE_EMAIL_DOMAIN: z.string(),
```
Comment (geclos) [inline packages/env/src/index.ts:74]: if this is not optional and you don't set them in web/gateway the production containers will fail

File: apps/infra/src/app/production/shared.ts [hunk 2]
```diff
@@ -118,6 +120,7 @@ export const environment = pulumi
       { name: 'GATEWAY_HOSTNAME', value: 'gateway.latitude.so' }, // This is the host gateway is available at
       { name: 'GATEWAY_SSL', value: 'true' },
       { name: 'LATITUDE_DOMAIN', value: 'latitude.so' },
+      { name: 'LATITUDE_EMAIL_DOMAIN', value: 'mail.latitude.so' },
```
Comment (geclos) [inline apps/infra/src/app/production/shared.ts:123]: we are changing domains for emails?

Comment (andresgutgon) [reply apps/infra/src/app/production/shared.ts:123]: Yes, already configured in Mailgun

File: packages/env/src/index.ts (hunk 1 above)
Comment (andresgutgon) [reply packages/env/src/index.ts]: Changed

File: packages/core/src/mailers/mailers/adapters/mailgun.ts [hunk 3]
```diff
@@ -13,12 +13,15 @@ export type MailgunResponse = {
 export default function createMailgunTransport({
   transportOptions,
 }: MailerOptions) {
-  const domain = env.LATITUDE_DOMAIN
-  const apiKey = env.MAILER_API_KEY
+  const domain = env.LATITUDE_EMAIL_DOMAIN
+  const apiKey = env.MAILGUN_MAILER_API_KEY
 
   if (!domain || !apiKey) return null
```
Comment (geclos) [inline packages/core/src/mailers/mailers/adapters/mailgun.ts:19]: should we raise instead of returning null if the domain env vars aren't defined?

**Pull request merged**